### PR TITLE
Use new updateRecommendations API for RM demo

### PR DESCRIPTION
**Pull request by msvinaykumar**
Bulk update results
+
Update Recommendation

### CODE REVIEW

**Comment by kusumachalasani** [comment]
Code related to the --days="" parameter wasn't merged.

File: remote_monitoring_demo/remote_monitoring_demo.sh [hunk 1]
```diff
@@ -40,7 +40,6 @@ function usage() {
 	echo "c = supports minikube and openshift cluster-type"
 	echo "d = duration of benchmark warmup/measurement cycles"
 	echo "p = expose prometheus port"
-	echo "u = Kruize UI Image. Default - quay.io/kruize/kruize-ui:<version as in package.json>"
```
Comment (kusumachalasani) [inline remote_monitoring_demo/remote_monitoring_demo.sh:43]: Need to keep this line.
echo "u = Kruize UI Image. Default - quay.io/kruize/kruize-ui:<version as in package.json>"

File: remote_monitoring_demo/remote_monitoring_demo.sh [hunk 2]
```diff
@@ -80,59 +79,52 @@ function kruize_install() {
 		exit -1
 	fi
 	pushd autotune >/dev/null
-	# Checkout mvp_demo to get the latest mvp_demo release version
-	git checkout mvp_demo >/dev/null 2>/dev/null
-
-	AUTOTUNE_VERSION="$(grep -A 1 "autotune" pom.xml | grep version | awk -F '>' '{ split($2, a, "<"); print a[1] }')"
-	# Kruize UI repo
-	KRUIZE_UI_REPO="quay.io/kruize/kruize-ui"
```
Comment (kusumachalasani) [inline remote_monitoring_demo/remote_monitoring_demo.sh:88]: UI related changes are missing.

File: remote_monitoring_demo/demo.py [hunk 3]
```diff
@@ -130,13 +142,39 @@ def main(argv):
...
+        file_path = './resource_usage_metrics_data/result_%s_to_%s.json' % (current_index, batch_size)
+        with open(file_path, 'w') as json_file:
+            json.dump(batch, json_file)
+        update_results(file_path)
+        batch_deep_copy = copy.deepcopy(batch)
+        for item in batch_deep_copy:
+            item['interval_start_time'] = datetime.strptime(item['interval_start_time'], "%Y-%m-%dT%H:%M:%S.%fZ")
+            item['interval_end_time'] = datetime.strptime(item['interval_end_time'], "%Y-%m-%dT%H:%M:%S.%fZ")
+        max_time = max(batch_deep_copy, key=lambda x: x['interval_end_time'])['interval_end_time']
+        update_recommendations(experiment_name, max_time.strftime("%Y-%m-%dT%H:%M:%S.%fZ")[:-4] + "Z")
```
Comment (dinogun) [inline remote_monitoring_demo/demo.py:171]: Can we suppress printing the recommendation JSON that is returned from this call. It adds too much info to the demo output.

File: remote_monitoring_demo/remote_monitoring_demo.sh (hunk 1 above)
Comment (msvinaykumar) [reply remote_monitoring_demo/remote_monitoring_demo.sh]: done

File: remote_monitoring_demo/remote_monitoring_demo.sh (hunk 2 above)
Comment (msvinaykumar) [reply remote_monitoring_demo/remote_monitoring_demo.sh:88]: added back

File: remote_monitoring_demo/demo.py [hunk 4]
```diff
@@ -95,16 +100,23 @@ def main(argv):
...
                 update_results(resultsjson_file)
-                
+
+                resultsjson = json.load(open(resultsjson_file))
+                for item in resultsjson:
+                    item['interval_start_time'] = datetime.strptime(item['interval_start_time'],
+                                                                    "%Y-%m-%dT%H:%M:%S.%fZ")
+                    item['interval_end_time'] = datetime.strptime(item['interval_end_time'], "%Y-%m-%dT%H:%M:%S.%fZ")
+                max_time = max(resultsjson, key=lambda x: x['interval_end_time'])['interval_end_time']
+                update_recommendations(experiment_name, max_time.strftime("%Y-%m-%dT%H:%M:%S.%fZ")[:-4] + "Z")
```
[the remark this answers is not in the document]
Comment (msvinaykumar) [reply remote_monitoring_demo/demo.py:115]: should be possible

File: remote_monitoring_demo/demo.py (hunk 3 above)
Comment (msvinaykumar) [reply remote_monitoring_demo/demo.py:171]: done

File: remote_monitoring_demo/remote_monitoring_demo.sh [hunk 5]
```diff
@@ -338,25 +335,27 @@ prometheus=0
...
                         visualize)
                                 visualize=1
                                 ;;
-			days=*)
-				DATA_DAYS=${OPTARG#*=}
-				;;
-			*)
-				;;
-		esac
-		;;
```
Comment (dinogun) [inline remote_monitoring_demo/remote_monitoring_demo.sh:358]: Why is this being removed?

File: remote_monitoring_demo/remote_monitoring_demo.sh [hunk 6]
```diff
@@ -382,8 +382,7 @@ while getopts o:c:d:prstu:-: gopts; do
 		KRUIZE_UI_DOCKER_IMAGE="${OPTARG}"
 		;;
 	*)
-		usage
-		;;
+			usage
```
Comment (dinogun) [inline remote_monitoring_demo/remote_monitoring_demo.sh:385]: tab not aligned

**Review by dinogun — APPROVED** [review]
LGTM

**Pull request merged**